ghcjs-src/Miso/Subscription/WebSocket.hs

```diff
@@ -151,48 +129,207 @@ data SocketState
   | CLOSED     -- ^ 3
   deriving (Show, Eq, Ord, Enum)
 
--- | Retrieves current status of `WebSocket`
-getSocketState :: IO SocketState
-getSocketState = do
-  Just ws <- readIORef websocket
-  toEnum <$> getSocketState' ws
+-- | Low-level websocket handle, wraps the raw JSVal.
+newtype WebSocket_ = WebSocket_ JSVal
+
+-- | Create a new low-level websocket.
+newWebSocket_ :: MisoString -> [MisoString] -> IO (Either MisoString WebSocket_)
+{-# INLINE newWebSocket_ #-}
```

Comment:
I’m a bit suspicious of all these `INLINE` pragmas. They tend to be a good way to ruin compile times.

Reply:
The original had these on many of the functions so I was just following the convention. I'm totally fine with removing these.